### ISSUE TITLE
Massive Bundle Size

### ISSUE THREAD

**Issue by GodOfGrandeur** [paralleldrive/react-feature-toggles#209]
Hey,

I've been looking into the package and noticed it has a 47.36kb footprint. This is mostly because of the use of curry method from lodash/fp/curry in configure-feature.js file. 

Because you're not building using rollup or webpack the entire library is pulled in because of the lack of tree shaking so you end up with a much inflated bundle size. This dependency is also missing in the package.json causing errors on bundlephobia:

https://bundlephobia.com/result?p=@paralleldrive/react-feature-toggles

There are a couple solutions to resolving this:
1. Make use of https://www.npmjs.com/package/lodash.curry instead of all of lodash, this will dramatically reduce bundle size and is a like for like replacement. 
2. I haven't checked this to see if it's a like for like replacement but https://www.npmjs.com/package/@ramda/curry is even smaller package and built on better programming principles.
3. Currying can be done using arrow functions which is more declarative and of course doesn't need a library to handle it. This will require a bit more work in adding new tests but is probably the best option. 

Doing so should reduce this package size significantly.

**Comment by ericelliott** [comment]
I suppose we could use this, instead:

```js
// Tiny, recursive autocurry
const curry = (
  f, arr = []
) => (...args) => (
  a => a.length === f.length ?
    f(...a) :
    curry(f, a)
)([...arr, ...args]);
```

But importing from `lodash/fp/curry` should not cause that kind of bloat. That's worth investigating deeper, and (if true) reporting to Lodash.

What are you using to bundle?

**Comment by GodOfGrandeur** [comment]
There's lots of blog posts about how lodash has massive bundle sizes. I'm not sure whats happening in this case but think the curry module is just badly packaged. I've attached a screenshot with it added to a standard create-react-app and source-map-explorer package added do the analysing. 

<img width="1667" alt="image" src="https://user-images.githubusercontent.com/913150/55350239-85ab8380-54b3-11e9-8ef0-7102b0c4c030.png">

**Comment by kennylavender** [comment]
I like the idea of using our own curry function so that we can avoid adding webpack and rollup. Looks like we are only using lodash in that one spot as well, so we can remove lodash as a dependency.

**Comment by ericelliott** [comment]
@kennylavender LGTM. Does my autocurry spell work? It's very tiny. =)

**Comment by kennylavender** [comment]
@ericelliott I am sure it will lol, Il create some unit tests to make sure it works & keeps working 👍

**Comment by GodOfGrandeur** [comment]
@kennylavender thanks for looking at this, have you been able to get anywhere with this yet?

**Comment by kennylavender** [comment]
@GodOfGrandeur thanks for the reminder, just created #211

**Comment by kennylavender** [comment]
Published version 2.1.1 after #211 ( oddly matches the version lol )

Bundle phobia now works:
https://bundlephobia.com/result?p=@paralleldrive/react-feature-toggles@2.1.1

So from 47.36kb to 3.1kb 👍
